### PR TITLE
Commit on branch advanced-login

### PR DESCRIPTION
**Pull request by hatcaiday**
ready

### CODE REVIEW

**Comment by hatcaiday** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -14,5 +15,23 @@ def digest string
...
+      update remember_digest: User.digest remember_token
+    end
+
+    def authenticated? remember_token
+      return false if remember_digest.nil?
+      BCrypt::Password.new remember_digest.is_password?remember_token
+    end
+
+    def forget
+      update_attribute(:remember_digest, nil)
```
Comment (huuphuc93) [inline app/models/user.rb]: Sử dụng update thay cho update_attribute

**Comment by huuphuc93** [comment]
commented

**Comment by hatcaiday** [comment]
ready

File: app/models/user.rb [hunk 2]
```diff
@@ -14,5 +15,23 @@ def digest string
...
+      update remember_digest: User.digest(remember_token)
+    end
+
+    def authenticated? remember_token
+      return false if remember_digest.nil?
+      BCrypt::Password.new remember_digest.is_password?remember_token
+    end
+
+    def forget
+      update :remember_digest, nil
```
Comment (huuphuc93) [inline app/models/user.rb]: update remember_digest: nil

File: app/views/static_pages/help.html.erb [hunk 3]
```diff
@@ -1,6 +1,6 @@
-<% provide :title, "help" %>
-<h1><%= t "help" %></h1>
+<% provide :title, ".help_title" %>
```
Comment (huuphuc93) [inline app/views/static_pages/help.html.erb]: t(".help_title")

**Comment by huuphuc93** [comment]
commented
comment nhỏ

**Comment by hatcaiday** [comment]
ready

**Comment by hatcaiday** [comment]
ready

**Comment by huuphuc93** [comment]
merged

**Pull request merged**